### PR TITLE
feature (workers): Add the user's client IP to a session's connection information

### PR DESCRIPTION
**Pull request by jimlambrt**
- feature: Add client ip to inbound request information
- feature (workers): Add user client IP to connection info
- feature (db): Add user_client_ip to session_connection
- feature (workers): add user client ip to ConnectConnection API
- feature (sessions): Add support for user client ip in connection info
- feature (workers): Add support for user client ip in tcp protocol

This PR is dependent on #1678 and cannot be merged until it's approved and merged.

### CODE REVIEW

File: internal/db/schema/migrations/oss/postgres/19/03_session.up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+begin;
+
+alter table session_connection
+  add column user_client_ip inet;
```
Comment (mgaffney) [inline internal/db/schema/migrations/oss/postgres/19/03_session.up.sql:4]: Can a `not null` constraint be added? For existing rows, I suggest setting the value to `::` (the [IPv6 Unspecified Address](https://en.wikipedia.org/wiki/IPv6_address#Unspecified_address)).

Comment (jimlambrt) [reply internal/db/schema/migrations/oss/postgres/19/03_session.up.sql:4]: ty!

**Comment by jimlambrt** [comment]
@mgaffney -> sorry about the rebase/force-push after your review.

**Comment by jimlambrt** [comment]
Closing this in favor of #1749 which will be easier to keep up to date without conflicts.

**Pull request closed without merging**